pylint/testutils/primer.py

```diff
@@ -0,0 +1,69 @@
+import os
+import shutil
+from pathlib import Path
+from typing import List, NamedTuple, Optional
+
+import git
+
+PRIMER_DIRECTORY_PATH: Path = Path(".pylint_primer_tests")
+
+
+class PackageToLint(NamedTuple):
+    """Represents data about a package to be tested during primer tests"""
+
+    url: str
+    """URL of the repository to clone"""
+
+    branch: str
+    """Branch of the repository to clone"""
+
+    commit: Optional[str]
+    """Commit hash to pin the repository on"""
+
+    @property
+    def clone_directory(self) -> str:
+        """Directory to clone repository into"""
+        return str(
+            PRIMER_DIRECTORY_PATH
+            / "/".join(self.url.split("/")[-2:]).replace(".git", "")
+        )
+
+    directories: str
+    """Directories within the repository to run pylint over"""
+
+    pylint_additional_args: List[str] = []
```

Comment:
Default values are only supported in `3.6.1`, but I don't think that matters here.

Reply:
Yes, #5171  makes it ok :)

Reply:
```suggestion
    pylint_additional_args: List[str]
```

Reply:
@Pierre-Sassoulas I think you missed this comment (the previous conversation was resolved).

Reply:
Strange, sorry !